lib/ancestry/optimizers/pg.rb

```diff
@@ -0,0 +1,27 @@
+module Ancestry
+  module Optimizers
+    module Pg
+      # Update descendants with new ancestry (before save)
+      def update_descendants_with_new_ancestry
```

Comment:
Also, could you look at the new way we do `update_descendants_with_new_ancestry` and see that it still looks like the best approach now.

I've moved over to using arrays instead of all the string munging we were doing before.

Think using the update clause to update all could be a great optimization.

Do wonder about objects already in memory.

Reply:
I can not use arrays because the generated SQL for `update_all` does not care about each descendant's ancestor_ids `new_ancestor_ids = path_ids + (descendant.ancestor_ids - path_ids_in_database)`. It just uses `regexp_replace` to update all descendants. So I have to use a string concatenation for `update_all`.

> Do wonder about objects already in memory.

descendants need to be reloaded to get new values. I'll update and test again.